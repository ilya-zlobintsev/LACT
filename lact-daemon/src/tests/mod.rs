use insta::assert_json_snapshot;

use crate::{config::Config, server::handler::Handler};
use std::{fs, path::PathBuf};

#[tokio::test]
async fn snapshot_everything() {
    let test_data_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("src/tests/data");

    for vendor_dir in fs::read_dir(test_data_dir).unwrap().flatten() {
        for device_dir in fs::read_dir(vendor_dir.path()).unwrap().flatten() {
            let test_key = format!(
                "{}/{}",
                vendor_dir.file_name().to_string_lossy(),
                device_dir.file_name().to_string_lossy()
            );

            let handler = Handler::with_base_path(&device_dir.path(), Config::default(), true)
                .await
                .unwrap();
            let mut device_info = handler
                .generate_snapshot_device_info()
                .into_values()
                .next()
                .unwrap();
            // Remove vulkan information, as it can only be fetched from the current running system and affects snapshot output
            device_info
                .get_mut("info")
                .unwrap()
                .as_object_mut()
                .unwrap()
                .remove("vulkan_info");

            assert_json_snapshot!(test_key, device_info);
        }
    }
}
