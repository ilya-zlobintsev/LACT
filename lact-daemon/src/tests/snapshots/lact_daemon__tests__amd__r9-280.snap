---
source: lact-daemon/src/tests/mod.rs
expression: device_info
---
{
  "clocks_info": null,
  "info": {
    "driver": "amdgpu",
    "link_info": {
      "current_speed": "5.0 GT/s PCIe",
      "current_width": "8"
    },
    "pci_info": {
      "device_pci_info": {
        "model": "Tahiti PRO [Radeon HD 7950/8950 OEM / R9 280]",
        "model_id": "679A",
        "vendor": "Advanced Micro Devices, Inc. [AMD/ATI]",
        "vendor_id": "1002"
      },
      "subsystem_pci_info": {
        "model": "GIGABYTE R9 280 WindForce 3X",
        "model_id": "229A",
        "vendor": "Gigabyte Technology Co., Ltd",
        "vendor_id": "1458"
      }
    },
    "vbios_version": "xxx-xxx-xxx",
    "vulkan_instances": []
  },
  "pci_info": {
    "device_pci_info": {
      "model": "Tahiti PRO [Radeon HD 7950/8950 OEM / R9 280]",
      "model_id": "679A",
      "vendor": "Advanced Micro Devices, Inc. [AMD/ATI]",
      "vendor_id": "1002"
    },
    "subsystem_pci_info": {
      "model": "GIGABYTE R9 280 WindForce 3X",
      "model_id": "229A",
      "vendor": "Gigabyte Technology Co., Ltd",
      "vendor_id": "1458"
    }
  },
  "power_profile_modes": null,
  "power_states": {
    "core": [],
    "vram": []
  },
  "stats": {
    "clockspeed": {
      "gpu_clockspeed": 300,
      "vram_clockspeed": 150
    },
    "fan": {
      "control_enabled": false,
      "pmfw_info": {},
      "pwm_current": 139,
      "pwm_max": 255,
      "pwm_min": 0
    },
    "power": {},
    "temps": {
      "edge": {
        "crit": 120.0,
        "crit_hyst": 90.0,
        "current": 37.0
      }
    },
    "voltage": {},
    "vram": {}
  }
}
